src/main/docker/gitlab-gitlabci.yml

```diff
@@ -0,0 +1,37 @@
+version: "3.6"
+services:
+    gitlab:
+        build: gitlab
+        volumes:
+            - artemis-gitlab-data:/var/opt/gitlab
+            - artemis-gitlab-logs:/var/log/gitlab
+            - artemis-gitlab-config:/etc/gitlab
+        hostname: 'HOSTNAME'
+        environment:
+            GITLAB_OMNIBUS_CONFIG: |
+                external_url 'https://EXTERNALURL'
+                prometheus_monitoring['enable'] = false
+                gitlab_rails['gitlab_shell_ssh_port'] = 2222
+                gitlab_rails['monitoring_whitelist'] = ['0.0.0.0/0']
+                letsencrypt['enable'] = true
+                letsencrypt['auto_renew_hour'] = "12"
+                letsencrypt['auto_renew_minute'] = "30"
+                letsencrypt['auto_renew_day_of_month'] = "*/7"
```

Comment:
I think you should exclude these from the base configuration. Let's encrypt requires the service to be available publicly. For local testing this is not the case. I suggest you add a [docker-compose override](https://docs.docker.com/compose/extends/) for production use and switch off HTTPS for local dev deployments.

Reply:
Thank you! I have just improved it.